include/eredis.hrl

```diff
@@ -54,6 +54,15 @@
 }).
 
 -define(NL, "\r\n").
+-define(NL_PATTERN,
+        case get(eredis_newline_pattern) of
+                undefined ->
+                    Pattern = binary:compile_pattern(?NL),
```

Comment:
It needs to be a binary. :)

```suggestion
                    Pattern = binary:compile_pattern(<<?NL>>),
```

Btw, it's risky to use variables in macros. This macro binds or matches Pattern in the scope where it's used.

I suggest instead use function and declare it as inline. See https://www.erlang.org/doc/man/compile.html#inlining

Reply:
Checking